googletest/include/gtest/internal/gtest-internal.h

```diff
@@ -1337,17 +1346,24 @@ constexpr bool InstantiateTypedTestCase_P_IsDeprecated() { return true; }
 
 #define GTEST_TEST_NO_THROW_(statement, fail) \
   GTEST_AMBIGUOUS_ELSE_BLOCKER_ \
-  if (::testing::internal::AlwaysTrue()) { \
+  if (::testing::internal::TrueWithString gtest_msg = std::string()) { \
     try { \
       GTEST_SUPPRESS_UNREACHABLE_CODE_WARNING_BELOW_(statement); \
     } \
+    catch (std::exception const& e) { \
+      gtest_msg.value = " std::exception-derived exception with description: "; \
+      gtest_msg.value += e.what(); \
+      gtest_msg.value += "."; \
+      goto GTEST_CONCAT_TOKEN_(gtest_label_testnothrow_, __LINE__); \
+    } \
     catch (...) { \
+      gtest_msg.value = "."; \
       goto GTEST_CONCAT_TOKEN_(gtest_label_testnothrow_, __LINE__); \
     } \
   } else \
     GTEST_CONCAT_TOKEN_(gtest_label_testnothrow_, __LINE__): \
-      fail("Expected: " #statement " doesn't throw an exception.\n" \
-           "  Actual: it throws.")
+      fail(("Expected: " #statement " doesn't throw an exception.\n" \
```

Comment:
`fail` won't take a `std::string`?

Reply:
No, it doesn't. It expands to 4-th parameter of `AssertHelper` c-tor: https://github.com/google/googletest/blob/90a443f9c2437ca8a682a1ac625eba64e1d74a8a/googletest/include/gtest/gtest.h#L1772